src/components/AddCityModal/AddCityModal.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useRef } from "react";
+import { Modal, Typography, Form, Input, Space } from "antd";
+
+export const AddCityModal = ({ isOpen, handleClose, handleAddCity }) => {
+  const [form] = Form.useForm();
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    if (!isOpen) {
+      form.resetFields();
+    } else {
+      setTimeout(() => inputRef?.current.focus(), 0);
+    }
+  }, [isOpen]);
+
+  const handleSubmit = () => {
+    form.submit();
+  };
+
+  const onFinish = (values) => {
+    handleAddCity(values["city-name"]);
+  };
+
+  return (
+    <Modal visible={isOpen} onCancel={handleClose} onOk={handleSubmit}>
+      <Space direction={"vertical"} align={"center"} style={{ width: "100%" }}>
+        <Typography>Enter city name</Typography>
+        <Form form={form} onFinish={onFinish}>
+          <Form.Item
+            name={"city-name"}
```

Comment:
it is better to use camel case for properties name. It will be easier to access them in javascript